ethstore/src/dir/disk.rs

```diff
@@ -161,14 +190,47 @@ impl KeyDirectory for DiskDirectory {
 	}
 
 	fn path(&self) -> Option<&PathBuf> { Some(&self.path) }
+
+	fn as_vault_provider(&self) -> Option<&VaultKeyDirectoryProvider> {
+		Some(self)
+	}
+}
+
+impl<T> VaultKeyDirectoryProvider for DiskDirectory<T> where T: KeyFileManager {
+	fn create(&self, name: &str, key: VaultKey) -> Result<Box<VaultKeyDirectory>, Error> {
+		let vault_dir = VaultDiskDirectory::create(&self.path, name, key)?;
+		Ok(Box::new(vault_dir))
+	}
+
+	fn open(&self, name: &str, key: VaultKey) -> Result<Box<VaultKeyDirectory>, Error> {
+		let vault_dir = VaultDiskDirectory::at(&self.path, name, key)?;
+		Ok(Box::new(vault_dir))
+	}
 }
 
+impl DiskKeyFileManager {
+	pub fn new() -> Self {
+		DiskKeyFileManager {}
```

Comment:
actually, `new` is unnecessary, it's less work to write `let _ = DiskKeyFileManager` than `let _ = DiskKeyFileManager::new()`